pkg/controller/eventstreamsadminv1/topic.go

```diff
@@ -0,0 +1,214 @@
+	opts, err := ibmc.GetAuthInfo(ctx, c.kube, mg)
+	if err != nil {
+		return nil, errors.Wrap(err, ibmc.ErrGetAuth)
+	}
+
+	service, err := c.clientFn(opts)
+	if err != nil {
+		return nil, errors.Wrap(err, ibmc.ErrNewClient)
+	}
+
+	return &topicExternal{client: service, kube: c.kube, logger: c.logger}, nil
+}
+
+// An topicExternal observes, then either creates, updates, or deletes an
+// external resource to ensure it reflects the managed resource's desired state.
+type topicExternal struct {
+	client ibmc.ClientSession
+	kube   client.Client
+	logger logging.Logger
+}
+
+func (c *topicExternal) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
+	cr, ok := mg.(*v1alpha1.Topic)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotTopic)
+	}
+
+	if meta.GetExternalName(cr) == "" {
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	instance, _, err := c.client.AdminrestV1().GetTopic(&arv1.GetTopicOptions{TopicName: reference.ToPtrValue(meta.GetExternalName(cr))})
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(ibmc.IsResourceNotFound, err), errGetTopicFailed)
+	}
+
+	currentSpec := cr.Spec.ForProvider.DeepCopy()
+	if err = ibmct.LateInitializeSpec(&cr.Spec.ForProvider, instance); err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, ibmc.ErrManagedUpdateFailed)
+	}
+	if !cmp.Equal(currentSpec, &cr.Spec.ForProvider) {
+		if err := c.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, ibmc.ErrManagedUpdateFailed)
+		}
+	}
+
+	cr.Status.AtProvider, err = ibmct.GenerateObservation(instance)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, ibmc.ErrGenObservation)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Available())
+
+	upToDate, err := ibmct.IsUpToDate(&cr.Spec.ForProvider, instance, c.logger)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, ibmc.ErrCheckUpToDate)
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:    true,
+		ResourceUpToDate:  upToDate,
+		ConnectionDetails: nil,
+	}, nil
+}
+
+func (c *topicExternal) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.Topic)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotTopic)
+	}
+
+	cr.SetConditions(runtimev1alpha1.Creating())
+	resInstanceOptions := &arv1.CreateTopicOptions{}
+	if err := ibmct.GenerateCreateTopicOptions(cr.Spec.ForProvider, resInstanceOptions); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateTopicOpts)
+	}
+
+	_, err := c.client.AdminrestV1().CreateTopic(resInstanceOptions)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateTopic)
+	}
+
+	// create topic doesn't return an instance so have to get one (unlike access group or resource key) ??
```

Comment:
Why do you need to get the instance here ? It should not be necessary. The instance should only be retrieved in the Observe() function. All you need to know when you create is if it failed or not. After the Create() the Crossplane runtime will take care behind the scenes to call Observe and check there for the status of the resource via Get(). Since there is no ID but only name for the Topic, you do not need instance.Name but can just use cr.Spec.ForProvider.Name to set the external name.